pycaret/internal/pycaret_experiment/tabular_experiment.py

```diff
@@ -2614,44 +2614,45 @@ def create_api(self, estimator, api_name, host="127.0.0.1", port=8000):
         _check_soft_dependencies("uvicorn", extra="mlops", severity="error")
         import uvicorn
 
-        MODULE = self._ml_usecase
-        INPUT_COLS = list(self.X.columns)
-        INPUT_COLS_WITHOUT_SPACES = [i.replace(" ", "_") for i in INPUT_COLS]
-        INPUT_COLS_WITHOUT_SPACES = [
-            i.replace("-", "_") for i in INPUT_COLS_WITHOUT_SPACES
-        ]
-        INPUT_COLS_WITHOUT_SPACES_FORMATTED = ", ".join(
-            tuple(INPUT_COLS_WITHOUT_SPACES)
-        ).replace("'", "")
+        _check_soft_dependencies("pydantic", extra="mlops", severity="error")
+        import pydantic
+
+        MODULE = (
+            self._ml_usecase.name.lower()
+        )  ## added .name.lower() as output changed from main branch
         API_NAME = api_name
         HOST = host
 
         self.save_model(estimator, model_name=api_name, verbose=False)
-
+        targetname = self.target_param + "_" + "prediction"
+        ## Removed tabs from query as that was causing the original file to have indentation errors.
         query = """
-    import pandas as pd
-    from pycaret.{MODULE_NAME} import load_model, predict_model
-    from fastapi import FastAPI
-    import uvicorn
-    # Create the app
-    app = FastAPI()
-    # Load trained Pipeline
-    model = load_model('{API_NAME}')
-    # Define predict function
-    @app.post('/predict')
-    def predict({INPUT_COLS}):
-        data = pd.DataFrame([[{DATAFRAME}]])
-        data.columns = {COLUMNS}
-        predictions = predict_model(model, data=data)
-        return {D1}'prediction': list(predictions['Label']){D2}
-    if __name__ == '__main__':
-        uvicorn.run(app, host='{HOST}', port={PORT})""".format(
+import pandas as pd
+from pycaret.{MODULE_NAME} import load_model, predict_model
+from fastapi import FastAPI
+import uvicorn
+from pydantic import create_model
+# Create the app
+app = FastAPI()
+# Load trained Pipeline
+model = load_model('{API_NAME}')
+# Create input/output pydantic models
+pydanticinputmodel=create_model('{API_NAME}_input',**{inputDataframeschema})
```

Comment:
can we format the query code for consistency?

Reply:
Ran black on the query and commited ab204b24 , however the code it generates won't be properly formatted because we are printing dictionairies. A possible workaround would be to print the ` json.dumps(inputDataframeschema,indent=4) `but I'm not sure if you wanted to go that far.